### PR TITLE
Improve docs regarding manifest usage

### LINKED ISSUE #307: Frequent WARN messages from Spring Cloud Services config server connector
When running the server I observe many WARN messages for the config server.  

```Tue Apr 18 2017
17:35:36.495
[APP/0]
OUT2017-04-18 21:35:36.487 WARN 13 --- [nio-8080-exec-5] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
[APP/0] OUT 2017-04-18 21:21:31.338 WARN 13 --- [io-8080-exec-10] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
Tue Apr 18 2017
17:21:31.338
[APP/0]
OUT2017-04-18 21:21:31.338 WARN 13 --- [io-8080-exec-10] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
[APP/0] OUT 2017-04-18 21:17:46.147 WARN 13 --- [nio-8080-exec-5] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
Tue Apr 18 2017
17:17:46.168
[APP/0]
OUT2017-04-18 21:17:46.147 WARN 13 --- [nio-8080-exec-5] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
[APP/0] OUT 2017-04-18 21:17:16.724 WARN 13 --- [nio-8080-exec-6] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
Tue Apr 18 2017
17:17:16.724
[APP/0]
OUT2017-04-18 21:17:16.724 WARN 13 --- [nio-8080-exec-6] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
[APP/0] OUT 2017-04-18 21:15:15.880 WARN 13 --- [nio-8080-exec-2] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
Tue Apr 18 2017
17:15:15.883
[APP/0]
OUT2017-04-18 21:15:15.880 WARN 13 --- [nio-8080-exec-2] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
[APP/0] OUT 2017-04-18 21:14:45.873 WARN 13 --- [nio-8080-exec-7] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
Tue Apr 18 2017
17:14:45.875
[APP/0]
OUT2017-04-18 21:14:45.873 WARN 13 --- [nio-8080-exec-7] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
[APP/0] OUT 2017-04-18 21:14:15.984 WARN 13 --- [nio-8080-exec-2] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
Tue Apr 18 2017
17:14:15.984
[APP/0]
OUT2017-04-18 21:14:15.984 WARN 13 --- [nio-8080-exec-2] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
[APP/0] OUT 2017-04-18 21:13:46.191 WARN 13 --- [nio-8080-exec-7] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
Tue Apr 18 2017
17:13:46.192
[APP/0]
OUT2017-04-18 21:13:46.191 WARN 13 --- [nio-8080-exec-7] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
[APP/0] OUT 2017-04-18 21:13:15.952 WARN 13 --- [io-8080-exec-10] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
Tue Apr 18 2017
17:13:15.952
[APP/0]
OUT2017-04-18 21:13:15.952 WARN 13 --- [io-8080-exec-10] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
[APP/0] OUT 2017-04-18 21:12:46.131 WARN 13 --- [nio-8080-exec-7] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
Tue Apr 18 2017
17:12:46.140
[APP/0]
OUT2017-04-18 21:12:46.131 WARN 13 --- [nio-8080-exec-7] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
[APP/0] OUT 2017-04-18 21:12:16.082 WARN 13 --- [nio-8080-exec-4] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
Tue Apr 18 2017
17:12:16.082
[APP/0]
OUT2017-04-18 21:12:16.082 WARN 13 --- [nio-8080-exec-4] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
[APP/0] OUT 2017-04-18 21:11:45.956 WARN 13 --- [nio-8080-exec-3] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
Tue Apr 18 2017
17:11:45.957
[APP/0]
OUT2017-04-18 21:11:45.956 WARN 13 --- [nio-8080-exec-3] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
[APP/0] OUT 2017-04-18 21:10:53.373 WARN 13 --- [nio-8080-exec-6] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
Tue Apr 18 2017
17:10:53.390
[APP/0]
OUT2017-04-18 21:10:53.373 WARN 13 --- [nio-8080-exec-6] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
[APP/0] OUT 2017-04-18 21:07:01.867 WARN 13 --- [ main] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
Tue Apr 18 2017
17:07:01.867
[APP/0]
OUT2017-04-18 21:07:01.867 WARN 13 --- [ main] c.c.c.ConfigServicePropertySourceLocator : Could not locate PropertySource: I/O error on GET request for "http://localhost:8888/mlp-scdf/cloud": Connection refused (Connection refused); nested exception is java.net.ConnectException: Connection refused (Connection refused)
```

However, I only see one WARN message for the deployed apps, which I believe also include the same Spring Cloud Services config server connector.  

Investigate how to get rid of these warnings since they are a false negative when one purposely doesn't want to connect to the SCS config server.  

Note, the library is included in order to facilitate an easy path to using the SCS tile, that is w/o having to rebuild the server.

### LINKED ISSUE #326: Impove documentation for use of manifest files
The current docs show entries for maven repo that look to be correct values, but are not.  Also, we can specify some additional values such as app name, memory and disk, so users can be closer to a cut-n-paste usage of the manifest.

### PR DESCRIPTION
**Pull request by markpollack**
* Document `SPRING_CLOUD_CONFIG_ENABLED=false` to suppress WARN messages when
you don't want to use the config server

Fixes #326
Fixes #307

### CODE REVIEW

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc [hunk 1]
```diff
@@ -229,29 +229,26 @@ Following is a sample template to provision the server on PCFDev.
...
     SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_DOMAIN: local.pcfdev.io
     SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_USERNAME: admin
     SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_PASSWORD: admin
-    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_STREAM_SERVICES: rabbit,redis
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_STREAM_SERVICES: rabbit
     SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_TASK_SERVICES: mysql
     SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_SKIP_SSL_VALIDATION: true
-    SPRING_APPLICATION_JSON {"maven": { "remote-repositories": { "repo1": { "url": "https://repo.spring.io/libs-snapshot"} } } }
-    JAVA_OPTS: '-Dlogging.level.cloudfoundry=DEBUG'
+    MAVEN_REMOTE_REPOSITORIES_REPO1_URL: https://repo.spring.io/libs-release
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc]: I thought that way of doing happened to work "by accident" (see https://github.com/spring-cloud/spring-cloud-dataflow-server-cloudfoundry/pull/306)

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc [hunk 2]
```diff
@@ -229,29 +229,26 @@ Following is a sample template to provision the server on PCFDev.
...
 services:
 - mysql
-- config-server
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc]: Is this removal on purpose? How is the **server** bound to config-server then?

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc [hunk 3]
```diff
@@ -546,7 +543,9 @@ If you're building your application from scratch and want to add the client side
 Where, `CONFIG_CLIENT_VERSION` can be the latest release of https://github.com/pivotal-cf/spring-cloud-services-connector/releases[Spring Cloud Config Server]
 client for Pivotal Cloud Foundry.
 
-Note that you will observe a `WARN` logging message if the application that uses this library can not connect to the server when the applicaiton starts and whenever the `/health` endpoint is accessed. 
+NOTE: You will observe a `WARN` logging message if the application that uses this library can not connect to the server when the applicaiton starts and whenever the `/health` endpoint is accessed.
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc]: how about "cannot connect to the **config** server"

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc [hunk 4]
```diff
@@ -546,7 +543,9 @@ If you're building your application from scratch and want to add the client side
...
+You can disable the client library if you know that you are not using config server functionality by setting the environment variable `SPRING_CLOUD_CONFIG_ENABLED=false`.
+Another more, drastic option, is to disable the platform health check with the environment variable `SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_STREAM_HEALTH_CHECK=none`
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc]: extra comma

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc [hunk 5]
```diff
@@ -557,31 +556,31 @@ at the runtime.
...
-  host: test-server
-  memory: 1G
-  disk_quota: 1G
+- name: data-flow-server
+  host: data-flow-server
+  memory: 2G
+  disk_quota: 2G
   instances: 1
-  path: spring-cloud-dataflow-server-cloudfoundry-VERSION.jar
+  path: {ABSOLUTE PATH TO SERVER UBER-JAR}
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc]: is it really "absolute"?

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc [hunk 6]
```diff
@@ -557,31 +556,31 @@ at the runtime.
...
   env:
-    SPRING_APPLICATION_NAME: test-server
-    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_URL: <URL>
-    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_ORG: <ORG>
-    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_SPACE: <SPACE>
-    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_DOMAIN: <DOMAIN>
-    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_USERNAME: <USER>
-    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_PASSWORD: <PASSWORD>
+    SPRING_APPLICATION_NAME: data-flow-server
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_URL: https://api.local.pcfdev.io
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc]: why switch to pcfdev explicit values all of a sudden?

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc [hunk 7]
```diff
@@ -557,31 +556,31 @@ at the runtime.
...
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_ORG: pcfdev-org
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_SPACE: pcfdev-space
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_DOMAIN: local.pcfdev.io
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_USERNAME: admin
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_PASSWORD: admin
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_STREAM_SERVICES: rabbit,my-config-server
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_TASK_SERVICES: mysql,my-config-server
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_SKIP_SSL_VALIDATION: true
     MAVEN_REMOTE_REPOSITORIES_REPO1_URL: https://repo.spring.io/libs-release
-    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_STREAM_SERVICES: my-config-server #this is for all the stream applications
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc]: The `#comments` were added to help users what goes where. Don't you think they clarify what happens?

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc [hunk 8]
```diff
@@ -557,31 +556,31 @@ at the runtime.
...
-    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_TASK_SERVICES: my-config-server #this is for all the task applications
 services:
 - mysql
-- my-config-server #this is for the server
+- my-config-server
 ----
 
-Where, `config-server` is the name of the Spring Cloud Config Service instance running on Cloud Foundry. By binding the
+Where, `my-config-server` is the name of the Spring Cloud Config Service instance running on Cloud Foundry. By binding the
 service to both Spring Cloud Data Flow server and as well as all the Spring Cloud Stream and Spring Cloud Task applications
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc]: "and as well as" is wrong

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc (hunk 1 above)
Comment (markpollack) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc]: It works, but it is not appropriate in a getting started section to have low level debug logging and it is better to point people to the release repo vs. snapshots where apps maybe unstable.

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc (hunk 2 above)
Comment (markpollack) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc]: Getting started does not cover using the config server.  There is another manifest later in the doc that has a manifest wrt to config server usage.

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc (hunk 3 above)
Comment (markpollack) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc]: Sure, can cut-n-paste the exact message shown which is listed in the GH issue.

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc (hunk 5 above)
Comment (markpollack) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc]: I'll find out, that is what was listed previously in the getting started section for the manifest.  I think relative also works.

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc (hunk 6 above)
Comment (markpollack) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc]: We have two manifest files in the docs, they were not consistent before this PR.  Now they are.  I suppose that we can add in more real world values using pcfdev?  I'm ok to switch it to PWS values as well.

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc (hunk 7 above)
Comment (markpollack) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc]: I notice now that one of the manifest examples has comments and the other don't.  When I was looking at it, it looked actually a bit distracting to me since the line length in the docs is then very long.  Not sure it adds a ton of value.  Can go either way, but we should be consistent.

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc (hunk 8 above)
Comment (markpollack) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc]: doh!

File: spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc [hunk 9]
```diff
@@ -557,31 +556,31 @@ at the runtime.
...
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_URL: https://api.local.pcfdev.io
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_ORG: pcfdev-org
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_SPACE: pcfdev-space
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_DOMAIN: local.pcfdev.io
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_USERNAME: admin
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_PASSWORD: admin
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_STREAM_SERVICES: rabbit,my-config-server
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_TASK_SERVICES: mysql,my-config-server
+    SPRING_CLOUD_DEPLOYER_CLOUDFOUNDRY_SKIP_SSL_VALIDATION: true
     MAVEN_REMOTE_REPOSITORIES_REPO1_URL: https://repo.spring.io/libs-release
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-cloudfoundry-docs/src/main/asciidoc/getting-started.adoc]: use SAJ

**Comment by ericbottard** [comment]
Edited and merged as 0968e35
